### PR TITLE
Add analyzer checks stats API

### PR DESCRIPTION
**Pull request by La0**
Refs #187

### CODE REVIEW

File: backend/code_review_backend/issues/tests/test_stats.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                mercurial_hash=hashlib.sha1(f"hg {i}".encode("utf-8")).hexdigest(),
+            )
+
+        # Create lots of issues randomly affected on diffs
+        analyzers = ["analyzer-X", "analyzer-Y", "analyzer-Z"]
+        checks = ["check-1", "check-10", "check-42", "check-1000", None]
+
+        for i in range(500):
+            Issue.objects.create(
+                diff_id=random.randint(1, 10),
```
Comment (marco-c) [inline backend/code_review_backend/issues/tests/test_stats.py]: Can you add a few Issues with no linked diff too, just to make sure that works too?

File: backend/code_review_backend/issues/api.py [hunk 2]
```diff
@@ -48,7 +52,25 @@ def perform_create(self, serializer):
...
 
+class IssueCheckStats(generics.ListAPIView):
+    """
+    List all analyzer checks per repository aggregated with
+    their total number of issues
+    """
+
+    serializer_class = IssueCheckSerializer
+    queryset = (
+        Issue.objects.values("analyzer", "check", "diff__revision__repository__slug")
```
Comment (marco-c) [inline backend/code_review_backend/issues/api.py]: What do we do for Issues not linked to diffs?

File: backend/code_review_backend/issues/tests/test_stats.py (hunk 1 above)
Comment (La0) [reply backend/code_review_backend/issues/tests/test_stats.py]: Done

File: backend/code_review_backend/issues/api.py (hunk 2 above)
Comment (La0) [reply backend/code_review_backend/issues/api.py]: repository falls back to None in the output ([see added unit test](https://github.com/mozilla/code-review/pull/190/commits/e41ad9bedb5ae9b1edca3be1b28737b1cc721d7d#diff-e22f97f7cbc51441508e5c7e52414aecR198))

Comment (marco-c) [reply backend/code_review_backend/issues/api.py]: Maybe we should have a direct link between issue and repository, as issues should not need a diff to be linked to a repository

**Pull request merged**